lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -271,6 +283,101 @@ def __get_k8s_job_spec_template(self, ajs):
+                eps = self.app.interactivetool_manager.configure_entry_points(ajs.job_wrapper.get_job(), ports_dict)
+                entry_points = []
+                for entry_point in eps.get('configured', []):
+                    log.debug(entry_point.to_dict())
+                    # sending in self.app as `trans` since it's only used for `.security` so seems to work
+                    entry_point_path = self.app.interactivetool_manager.get_entry_point_path(self.app, entry_point)
+                    if '?' in entry_point_path:
+                        # Removing all the parameters from the ingress path, but they will still be in the database
+                        # so the link that the user clicks on will still have them
+                        log.warn("IT urls including parameters (eg: /myit?mykey=myvalue) are only experimentally supported on K8S")
+                        entry_point_path = entry_point_path.split('?')[0]
+                    entry_point_domain = f'{self.app.config.interactivetools_proxy_host}'
+                    if entry_point.requires_domain:
+                        entry_point_subdomain = self.app.interactivetool_manager.get_entry_point_subdomain(self.app, entry_point)
+                        entry_point_domain = f'{entry_point_subdomain}.{entry_point_domain}'
+                    entry_points.append({"tool_port": entry_point.tool_port, "domain": entry_point_domain, "entry_path": entry_point_path})
+            k8s_spec_template = {
+                "metadata": {
+                    "labels": {
+                        "app.galaxyproject.org/handler": self.__force_label_conformity(self.app.config.server_name),
+                        "app.galaxyproject.org/destination": self.__force_label_conformity(
+                            str(ajs.job_wrapper.job_destination.id))
+                    },
+                    "annotations": {
+                        "app.galaxyproject.org/tool_id": ajs.job_wrapper.tool.id
+                    }
+                },
+                "spec": {
+                    "rules": [{"host": ep["domain"],
+                               "http": {
+                                   "paths": [{
+                                       "backend": {
+                                           "serviceName": "{}-{}".format(self.__produce_k8s_job_prefix(),
+                                                                         self.__force_label_conformity(ajs.job_wrapper.get_id_tag())),
+                                           "servicePort": int(ep["tool_port"])
+                                           # "service": {
+                                           #     "name": "job-{}-{}".format(self.__force_label_conformity(ajs.job_wrapper.get_id_tag()), p),
+                                           #     "port": { "number": int(p)}
+                                           # }
```

Comment:
Do you still need this ?
```suggestion
                                           # "service": {
                                           #     "name": "job-{}-{}".format(self.__force_label_conformity(ajs.job_wrapper.get_id_tag()), p),
                                           #     "port": { "number": int(p)}
                                           # }
```

Reply:
Yes, forgot to change it back with the new api version. changed in https://github.com/galaxyproject/galaxy/pull/11383/commits/ec026579a5f6fafeb30c1212f52957586d484904